### DISCUSSION TITLE
Maintain your own configuration on nixos

### DISCUSSION THREAD

**Discussion by solitudealma** [ayamir/nvimdots#1258] [General]
I forked this repo. What I wanted to do was to add my own stuff to user_template first, then change the url to a local path, build the nixos configuration and test it, and then upload it to the fork repo. Is there anything that needs attention in this process, such as the repo itself not processing this directory or .gitkeep affecting the nixos build? @misumisumi

**Comment by charliie-dev** [comment]
make sure when it't not in the gitignore when u update your own fork to github.

also, .gitkeep won't effect anything else, the only purpose of .gitkeep is to keep an empty dir while pushing to github.

**Reply by charliie-dev** [reply]
u can also use 'nix develop' to test your newly-added plugins first instead of changing the remote url to a local directory.

**Reply by solitudealma** [reply]
> u can also use 'nix develop' to test your newly-added plugins first instead of changing the remote url to a local directory.

sry,I didn't see this reply, I'll try it first. ask again if encounter any problems.

**Comment by solitudealma** [comment]
> make sure when it't not in the gitignore when u update your own fork to github.
> 
> also, .gitkeep won't effect anything else, the only purpose of .gitkeep is to keep an empty dir while pushing to github.

Do I have to update to github? Because I am not sure that this change is correct, so I want to test it without submitting it, and then push it to the repo after passing it. Can this be achieved?

**Comment by misumisumi** [comment]
You can add your own files by using `git add --force` without updating .gitignore.
Since `homeManagerModule` is provided, it can be easily implemented using `home-manager`.
wiki is too old, so you can see how to setting in #1250

**Comment by misumisumi** [comment]
If you want to contribute to `ayamir/nvimdots`, create a new branch as follows
Never include your own configuration.
```sh
git remote add upstrem <repo-url>.
git checkout upstrem -b <branch_name>
```

**Comment by solitudealma** [comment]
> You can add your own files by using `git add --force` without updating .gitignore. Since `homeManagerModule` is provided, it can be easily implemented using `home-manager`. wiki is too old, so you can see how to setting in #1250

thx, is this the only solution now? Use nix develop in the nivmdots repository to test whether the added things are available. Then update to the remote. Instead of building the system directly in the normal environment and testing it as I said.

I saw that you also forked this repository and maintained it yourself. But you give up the user_template directory and handled the conflicts yourself. I don't want to handle conflicts myself while enjoying upstream updates, I want to find a better local testing solution. After testing, submit it to the remote repo

**Reply by misumisumi** [reply]
> Instead of building the system directly in the normal environment and testing it as I said.

Please make flake input url like `path:/absolute/path/to/nvimdots`, if you want to test without using `nix develop`.
Now, you can effect to change config with `nixos-rebuild (or home-manager) switch`.

> you give up the user_template directory and handled the conflicts yourself.

No need to rm `user_templates`.
You only need to create `user` dir under `lua`.
Please see [wiki/Usage](https://github.com/ayamir/nvimdots/wiki/Usage).

**Reply by solitudealma** [reply]
> Please make flake input url like `path:/absolute/path/to/nvimdots`, if you want to test without using `nix develop`.
Now, you can effect to change config with `nixos-rebuild (or home-manager) switch`.

Yes, I have used absolute paths or relative paths to point to the cloned warehouse, which contains my newly added configuration. But when nix-rebuild switch, the configuration I wanted was not generated, which means that except for the default user_template, the new things I added are not there. My main doubt now is here. Logically speaking, it should be generated. I don’t know much about nix syntax, so i only understand it a little bit.

> No need to rm user_templates.
You only need to create user dir under lua.
Please see [wiki/Usage](https://github.com/ayamir/nvimdots/wiki/Usage).

i know how to use, i just don't understand the above.

**Reply by solitudealma** [reply]
you can use my [repo](https://github.com/solitudealma/nvimdots) which is on the github or add what i add. then you can find problems i encountered

**Reply by misumisumi** [reply]
> i know how to use, i just don't understand the above.

After creating the user dir with `cp -r user_template user`, change the files under the `user` dir.
